sympy/stats/stochastic_process_types.py

```diff
@@ -0,0 +1,230 @@
+from sympy import (Symbol, Matrix, MatrixSymbol, S, Indexed, Basic,
+                    Set, And, Tuple, Eq, FiniteSet, ImmutableMatrix)
+from sympy.core.compatibility import string_types
+from sympy.stats.rv import RandomIndexedSymbol, random_symbols, RandomSymbol
+from sympy.stats.symbolic_probability import Probability
+from sympy.stats.stochastic_process import StochasticPSpace
+
+__all__ = [
+    'StochasticProcess',
+    'DiscreteMarkovChain'
+]
+
+class StochasticProcess(Basic):
+    """
+    Base class for all the stochastic process whether
+    discrete or continuous.
+
+    Parameters
+    ==========
+
+    sym: Symbol or string_types
+    state_space: Set
+        The state space of the stochastic process, by default S.Reals.
+        For discrete sets it is zero indexed.
+    """
+
+    is_Continuous = None
+    is_Discrete = None
+
+    def __new__(cls, sym, state_space=S.Reals):
+        if isinstance(sym, string_types):
+            sym = Symbol(sym)
+        if not isinstance(sym, Symbol):
+            raise TypeError("Name of stochastic process should be either only "
+                            "a string or Symbol.")
+        if isinstance(state_space, set):
+            state_space = list(state_space)
+        if isinstance(state_space, (list, tuple)):
+            state_space = FiniteSet(*state_space)
```

Comment:
No need to convert the set to a list before converting it to a FiniteSet: you can pass the set straight to FiniteSet.

Reply:
There is a lot of repetitive boiler-plate in this code. Is there not a helper function somewhere for coercing tuple/list/set/Set to Set?

Reply:
> Is there not a helper function somewhere for coercing tuple/list/set/Set to Set?

I don't think so. I will make one.